src/transformers/data/data_collator.py

```diff
@@ -138,6 +138,73 @@ def mask_tokens(self, inputs: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]
         # The rest of the time (10% of the time) we keep the masked input tokens unchanged
         return inputs, labels
 
+@dataclass
+class DataCollatorForAlbertPretrain(DataCollatorForLanguageModeling):
+    """
+    Data collator used for Albert Pretraining.
+    - collates batches of tensors, honoring their tokenizer's pad_token
+    - preprocesses batches for both masked language modeling and sentence order prediction
+    """
+
+    def __call__(self, examples: List[Dict[str, torch.tensor]]) -> Dict[str, torch.Tensor]:
+        token_list = []
+        token_type_list = []
+        sop_label_list = []
+        for example in examples:
+            token_list.append(example["tokens"])
+            token_type_list.append(example["segment_ids"])
+            sop_label_list.append(example["sentence_order_label"])
```

Comment:
Can we do those three as a list comprehension? This is going to be faster than a regular PyThon loop and this part is called for each batch, so we need speed. Here is one example:
```
token_list = [example["tokens"] for example in examples]
```

Second question: could we use `input_ids` and `token_type_ids` instead of tokens and segment_ids? That will surprise any use of the library less since we use those names for all tokenizers.

Reply:
already changed.thanks.